From f3019a82c703f7d0b30aa1456b11d821fa8a08f7 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 03 Feb 2022 09:59:41 +0200
Subject: [PATCH] [SW-0] fit: fix loadaddr print

64bit printing address wasn't working correctly and printed only lower
32 bit, changed to use puts function that's working.

Signed-off-by: Noam Lessner <nlessner@habana.ai>
Change-Id: Iebd9feac37681ae5ba06528a399e986a249c6290
---

diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 82b37de..e067c95 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -628,10 +628,7 @@
 static int fit_copy_image(u64 dst, const void *buf, int len, const char *arch,
 			  const char *cmpr)
 {
-	preboot_puts_raw("FIT loadaddr: 0x");
-	if (sizeof(void *) > sizeof(u32))
-		preboot_puth_raw((u32)(dst >> 32), false);
-	preboot_puth((u32)dst, false);
+	preboot_puts_hex64("FIT loadaddr: ", dst, false);
 
 	if (!strcmp(cmpr, "none")) {
 		preboot_copy_image(dst, buf, len, arch);
